### PR TITLE
Set up associates

### PR DESCRIPTION
**Pull request by Bavon101**
Hello, as per this PR I achieved the following:  

### Project requirements
- [x] Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- [x] Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- [x] Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- [x] Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- [x] Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (t-yanick) [inline book.rb]: - [ OPTIONAL] For better code organization, it will be nice to move these variables to the top of the class.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+end
```
Comment (t-yanick) [inline book.rb:15]: - [x] To implement the `has-many` side of the association between `Book` and `Rental`,  a method for adding rentals should be present in the `Book` class. Some like so:
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
Please kindly fix that.

File: person.rb [hunk 3]
```diff
@@ -8,10 +8,11 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (t-yanick) [inline person.rb]: - [x] To implement the `has-many` side of the association between `Person` and `Rental`,  a method for adding rentals should be present in the `Person` class. Some like so:
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```
Please kindly fix that.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    book.rentals << self
+    @person = person
+    person.rentals << self
+  end
+
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (t-yanick) [inline rental.rb]: - [ OPTIONAL] For better code organization, it will be nice to move these variables to the top of the class.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Bavon101, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kingstalux — APPROVED** [review]
Hi @Bavon101,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

## Highlights 👏 
✅ Correct Gitflow used 👍 
✅ Linters are passing 👏 
✅ Associations correctly implemented 🥂 
✅ Great pull request description 🔥 

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**